source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -207,3 +225,511 @@ void AMContinuousScanActionController::fillDataMaps(AMAgnosticDataAPIDataAvailab
+		qDebug() << "Scaler start movment index was out of bounds at " << scalerInitiateMovementIndex;
+		scalerInitiateMovementIndex += ((encoderUpVector_.count()-scalerInitiateMovementIndex)-expectedDurationScaledToBaseTimeScale_);
+		scalerRangeData.setMotionStartIndex(scalerInitiateMovementIndex);
+		qDebug() << "Changed to " << scalerInitiateMovementIndex;
+	}
+	if(!amptekDetectors_.isEmpty()){
+		AMDSClientDataRequest *oneAmptekRequest = clientDataRequestMap_.value(amptekDetectors_.first()->amdsBufferName());
+		if( (oneAmptekRequest->data().count() - amptekRangeData.motionStartIndex()) < expectedDurationScaledToBaseTimeScale_){
+			qDebug() << "Amptek start movment index was out of bounds at " << amptekRangeData.motionStartIndex();
+			amptekRangeData.setMotionStartIndex( ((oneAmptekRequest->data().count() - amptekRangeData.motionStartIndex()) - expectedDurationScaledToBaseTimeScale_) + amptekRangeData.motionStartIndex() );
+			qDebug() << "Changed to " << amptekRangeData.motionStartIndex();
+		}
+	}
+
+	foreach(CLSAMDSScalerChannelDetector *scalerChannelDetector, scalerChannelDetectors_)
+		detectorStartMotionIndexMap_.insert(scalerChannelDetector->name(), scalerRangeData.motionStartIndex());
+	if(!amptekDetectors_.isEmpty()){
+		foreach(CLSAmptekSDD123DetectorNew* amptekDetector, amptekDetectors_){
+			detectorStartMotionIndexMap_.insert(amptekDetector->name(), amptekRangeData.motionStartIndex());
+		}
+	}
+
+	qDebug() << "Amptek: " << amptekRangeData.motionStartIndex() << amptekRangeData.motionEndIndex() << amptekRangeData.listIndex();
+	qDebug() << "Scaler: " << scalerRangeData.motionStartIndex() << scalerRangeData.motionEndIndex() << scalerRangeData.listIndex();
+
+	scalerInitiateMovementIndex_ = detectorStartMotionIndexMap_.value(scalerChannelDetectors_.first()->name());
+
+	return true;
+}
+
+bool AMContinuousScanActionController::generateAxisFeedbackValues()
+{
+	return false;
+}
+
+bool AMContinuousScanActionController::generateInterpolatedParameters()
+{
+	double startAxisValue = axisFeedbackValues_.first();
+	double endAxisVaue = axisFeedbackValues_.last();
+	if(startAxisValue > endAxisVaue)
+		qSwap(startAxisValue, endAxisVaue);
+
+	qDebug() << "Start energy was " << startAxisValue << " end energy was " << endAxisVaue;
+	startAxisValue = ceilf(startAxisValue * (1/resolutionStep_)) / (1/resolutionStep_);
+	endAxisVaue = floorf(endAxisVaue * (1/resolutionStep_)) / (1/resolutionStep_);
+	qDebug() << "Start energy is " << startAxisValue << " end energy is " << endAxisVaue;
+
+	interpolatedSize_ = (endAxisVaue-startAxisValue)/resolutionStep_;
+	qDebug() << "interpolatedSize is " << interpolatedSize_;
+
+	// Create the interpolated axis and midpoints
+	double currentAxisValue = startAxisValue;
+	interpolatedAxis_ = QVector<double>(interpolatedSize_);
+	interpolatedMidpoints_ = QVector<double>(interpolatedSize_);
+	for(int x = 0, size = interpolatedAxis_.count(); x < size; x++){
+		interpolatedAxis_[x] = currentAxisValue;
```

Comment:
This one could be a memcpy or just use `interpolatedAxis_ = QVector<double>(interpolatedSize_, currentAxisValue);`.